internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -1443,8 +1443,34 @@ func flattenKubernetesClusterDataSourceKubeConfigAAD(config kubernetes.KubeConfi
 	return []interface{}{values}
 }
 
-func flattenClusterDataSourceIdentity(input *identity.SystemOrUserAssignedMap) (*[]interface{}, error) {
-	return identity.FlattenSystemOrUserAssignedMap(input)
+func flattenClusterDataSourceIdentity(input *managedclusters.ManagedClusterIdentity) (*[]interface{}, error) {
+	if input == nil {
+		return nil, nil
+	}
+	identityIds := make(map[string]identity.UserAssignedIdentityDetails)
+	if input.UserAssignedIdentities != nil {
+		for k, v := range *input.UserAssignedIdentities {
+			identityIds[k] = identity.UserAssignedIdentityDetails{
+				PrincipalId: v.PrincipalId,
+				ClientId:    v.ClientId,
+			}
+		}
+	}
+
+	out := identity.SystemOrUserAssignedMap{
+		IdentityIds: identityIds,
+	}
+	if input.Type != nil {
+		out.Type = identity.Type(*input.Type)
+	}
+	if input.PrincipalId != nil {
+		out.PrincipalId = *input.PrincipalId
+	}
+	if input.TenantId != nil {
+		out.TenantId = *input.TenantId
+	}
+
+	return identity.FlattenSystemOrUserAssignedMap(&out)
 }
```

Comment:
This shouldn't be necessary anymore since a fix for this was pushed to Pandora
https://github.com/hashicorp/pandora/pull/3614
```suggestion
```